delivery/client.py

```diff
@@ -109,9 +123,93 @@ class DeliveryServiceClient:
     def __init__(
         self,
         routes: DeliveryServiceRoutes,
+        cfg_factory=None,
     ):
         self._routes = routes
+        self.cfg_factory = cfg_factory or ci.util.ctx().cfg_factory()
```

Comment:
I would actually prefer to require the Client's initialiser to accept GitHub-Cfg-Elements (or even bearer-tokens) + add convenience functionality to util functions (e.g. in ccc.delivery)